packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1154,10 +1210,31 @@ async function getGCStateFromSnapshot(
         }
         const gcState = await readAndParseBlob<IGarbageCollectionState>(blobId);
         assert(gcState !== undefined, 0x2ad /* "GC blob missing from snapshot" */);
+        // Update latest serialized GC blobs
+        if (trackGCBlobState) {
+            serializedGCBlobs.set(key, JSON.stringify(generateSortedGCState(gcState)));
+        }
         // Merge the GC state of this blob into the root GC state.
         rootGCState = concatGarbageCollectionStates(rootGCState, gcState);
     }
-    return rootGCState;
+    return [rootGCState, serializedGCBlobs];
+}
+
+function generateSortedGCState(gcState: IGarbageCollectionState): IGarbageCollectionState {
+    const sortableArray: [string, IGarbageCollectionNodeData][] = Object.entries(gcState.gcNodes);
+    sortableArray.sort(([a], [b]) => a.localeCompare(b));
+    sortableArray.forEach(([, nodeData]) => {
```

Comment:
You can avoid this loop by sorting this while copying the state to the sorted state in the below for loop.